src/components/Auth/Login.js

```diff
@@ -1,19 +1,177 @@
 import Switch from 'components/Shared/Switch';
 import StyledButton from 'components/Shared/Button';
-import PwdInput from 'components/Auth/Shared/PwdInput';
-import IdInput from 'components/Auth/Shared/IdInput';
 import * as S from 'components/Auth/styles';
 import { login } from '../../store/auth';
 import { useNavigate } from 'react-router';
 
+const LoginContainer = styled.div`
+  display: flex;
+  justify-content: center;
+  flex-wrap: wrap;
+`;
+
+const LoginOptionContainer = styled.div`
+  margin-bottom: 40px;
+  border-bottom: 1px solid #eee;
+  width: 343px;
+`;
+
+const LoginOptionButton = styled.button`
+  min-width: 167px;
+  max-width: 176px;
+  padding: 12px 40.5px;
+  background: none;
+  font-size: 16px;
+  font-weight: ${({ isClicked }) => (isClicked ? 'bold' : 'normal')};
+  text-align: center;
+`;
+
+const LoginOptionMenuBar = styled.div`
+  display: block;
+  position: absolute;
+  width: ${({ location }) => (location === 1 ? 167 : 176)}px;
+  height: 1px;
+  background-color: #222;
+  transition: transform 0.2s ease;
+  transform: translateX(${({ location }) => (location === 1 ? 0 : 167)}px);
+`;
+
+const LoginForm = styled.form`
+  display: ${({ display }) => (!display ? 'none' : 'block')};
+  position: relative;
+  width: 100%;
+`;
+
+const StyledInput = styled(S.StyledInput)`
+  & + & {
+    margin-top: 0;
+  }
+`;
+
+const StyledOptionLink = styled(S.StyledLink)`
+  padding: 0 15px;
+  height: 12px;
+  :nth-child(n) {
+    border-right: 1px solid #999;
+  }
+  :last-child {
+    border: none;
+  }
+`;
+
+const StyledInputContainer = styled.div`
+  margin-bottom: 16px;
+  position: relative;
+  :nth-child(2) {
+    margin-bottom: 8px;
+  }
+`;
+
+const PwdSee = styled.span`
+  display: block;
+  right: 16px;
+  top: 14px;
+  position: absolute;
+  cursor: pointer;
+`;
+const EyeImg = styled.img`
+  width: 24px;
+  height: 24px;
+`;
+
+const SNSLoginOptionSection = styled.section`
+  display: ${({ display }) => (display ? 'none' : 'flex')};
+  width: 100%;
+  flex-wrap: wrap;
+  align-items: center;
+  justify-content: center;
+`;
+
+const GoogleLoginButton = styled.button`
+  width: 343px;
+  height: 44px;
+  margin-bottom: 16px;
+  padding: 0 10px;
+  border: solid 0.5px #c4c4c4;
+  background-color: #fff;
+
+  font-size: 16px;
+  font-weight: normal;
+  text-align: center;
+  color: #000;
+
+  background-image: url('/asset/google-logo.svg');
+  background-position-y: center;
+  background-position-x: 12px;
+  background-repeat: no-repeat;
+
+  :after {
+    content: '구글 로그인';
+  }
+`;
+const NaverLoginButton = styled.button`
+  width: 343px;
+  height: 44px;
+  margin-bottom: 16px;
+  padding: 0 0 4px;
+  background-color: #03c75a;
+
+  font-size: 16px;
+  font-weight: normal;
+  text-align: center;
+  color: #fff;
+
+  background-image: url('/asset/naver-logo.svg');
+  background-position-y: center;
+  background-position-x: 12px;
+  background-repeat: no-repeat;
+  :after {
+    content: '네이버 로그인';
+  }
+`;

```

Comment:
요렇게 할 수도 있어보이고요...!
[해당 문서](https://styled-components.com/docs/api#css) 참고해주세요

```suggestion
import { css } from "styled-components"

const commonButtonAttributes = css`
  width: 343px;
  height: 44px;
  margin-bottom: 16px;

  font-size: 16px;
  font-weight: normal;
  text-align: center;

  background-position-y: center;
  background-position-x: 12px;
  background-repeat: no-repeat;
`;

const GoogleLoginButton = styled.button`
  ${commonButtonAttributes};
  
  // ...
`

const NaverLoginButton = styled.button`
  ${commonButtonAttributes};
  
  // ...
`

```

Reply:
오 요런 방식도 있었군요 css를 그대로 쓸 수도 있었다니 ...